packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -709,3 +717,99 @@ class _CupertinoEdgeShadowPainter extends BoxPainter {
     canvas.drawRect(rect, paint);
   }
 }
+
+class _CupertinoModalPopupRoute<T> extends PopupRoute<T> {
+  _CupertinoModalPopupRoute({
+    this.builder,
+    this.barrierLabel,
+    RouteSettings settings,
+  }) : super(settings: settings);
+
+  final WidgetBuilder builder;
+
+  @override
+  final String barrierLabel;
+
+  @override
+  Color get barrierColor => _kModalBarrierColor;
+
+  @override
+  bool get barrierDismissible => true;
+
+  @override
+  bool get semanticsDismissible => false;
+
+  @override
+  Duration get transitionDuration => _kModalPopupTransitionDuration;
+
+  Animation<double> _animation;
+
+  @override
+  Animation<double> createAnimation() {
+    assert(_animation == null);
+    _animation = new CurvedAnimation(
+      parent: super.createAnimation(),
+      curve: Curves.ease,
+      reverseCurve: Curves.ease.flipped,
+    );
+    return _animation;
+  }
+
+  @override
+  Widget buildPage(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation) {
+    return builder(context);
+  }
+
+  @override
+  Widget buildTransitions(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget child) {
+    if (_animation.value == 1.0) {
+      context.findRenderObject().markNeedsSemanticsUpdate();
```

Comment:
buildTransition isn't guaranteed to be called each time the animation value changes.

Reply:
I've found in practice that the animation is long enough and the curve is such that buildTransitions ends up being called several times while the animation value is 1.0. I understand the concern though - could the markNeedsSemanticsUpdate() be called in buildPage, or do you know of another way to tell the render object there needs to be a semantics update when the animation is complete?

Reply:
What about dismissal? Don't you want to call it as soon as it's not 1.0 anymore? Add animation status listeners on createAnimation. (Make sure to unregister the listener when it's done though)

Reply:
actually why do you call this here in the route itself? What is the issue you're solving?

Reply:
I see. I would leave this out altogether. If it's easy to produce a reduced test case that demonstrates that the semantic box is at the wrong place inside a fractional translation, this should be fixed in the semantics pipeline itself. 

I'd leave it broken and leave an issue describing the misbehavior.